### PR TITLE
Make `--first-parent` an option

### PR DESCRIPTION
**Pull request by Tom-van-Woudenberg**
As described in #100. But having issues with the parsing of the output now...

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -51,14 +51,33 @@ Options
...
-    * By default, only the first-parent history is followed when determining
-      the last updated date (using ``git log --first-parent``).
-      This means that merges from the current branch into other branches
-      are ignored, and only changes merged into the current branch are
-      considered.  This is typically the desired behavior for documentation
-      builds, as it reflects when content actually entered the branch being
-      built.  If you want to follow the full history including all merge
-      parents, set ``git_first_parent = False`` in your ``conf.py``.
+    * By default, the full Git history is followed when determining
+      the last updated date, including all merge parents.
```
Comment (mgeier) [inline README.rst]: "the full history is followed" sounds confusing to me. This sounds as if in the other options parts of the history are ignored. I don't think that's true, because no change is ever ignored. The only difference is which date is assigned.

**Comment by Tom-van-Woudenberg** [comment]
> OK, this is horribly complicated ... I had to try it out ...

Yes it is ;)

> As far as I understand the observed behavior, I think it's important to note that merge commits are split into two parts (might be different with more than two parents, but let's not get off track here). This means that when using `-m`, I'm seeing two entries for each merge commit, one containing the changes from the first parent, the other the ones from the second (this is something I didn't expect). This in turn means that any change in any branch that is merged later in the history (regardless whether it was merged _into_ or _from_ the other branch) will be shown with the date from the merge commit. I think this is a behavior that nobody wants.

Agree

> Another thing that I found out is that `--first-parent -m` does the same as `--first-parent` (without the `-m`), as far as I could observe.
> If I'm understanding it right, I think we don't want to provide an option to enable `-m`

Agree.

> Using `--first-parent` doesn't mean the changes from other branches are ignored (which would have been surprising), but they are shown in the merge commits instead of where they actually happened. Changes from the `main` branch (or more precisely the "merged-into branch") are _not_ shown in merge commits, but only in the commits where they originally happened. I think this is the behavior you originally wanted (and probably had in mind for #99), and I think this is a reasonable option to add.
> If I'm understanding it right, I think we don't want to provide an option to enable `-m`, I think toggling `--first-parent` is all we need.

The intented behaviour of #99 was to 'ignore' merges from the current branch into other branches, not 'ignoring' merges from other branches into the current branch. So from a main perspective: a merge from main into a feature branch which is later merged back into main is ignored. Do we agree on that?

https://github.com/mgeier/sphinx-last-updated-by-git/pull/101#discussion_r2696018840:
> This only true when `-m` is used, right? If we don't provide `-m`, this problem should go away.

So yes, the `--first-parent` was only requried because of `-m`. If `-m` is not used we could skip this option since we all merge commits won't have files associated to it. So we could consider not having both `-m` and `--first-parent` at all. Keeping `--first-parent`  would allow users who configured `git_first_parent = True` (since the merge of #99 ) explicitly to keep the current behaviour. But I think we can break that behaviour because it was only there for a short time if we continue with this.

**Comment by Tom-van-Woudenberg** [comment]
> So, do we agree?

Yes

> I'm not sure if calling the option something with "first parent" helps understanding. What about something like `git_last_updated_when_merged = True`?

I agree that that makes more sense than 'first parent'!

> I have added some commits to the test repo (https://github.com/mgeier/sphinx-last-updated-by-git/pull/107) to have a way of unequivocally testing the "first parent" feature. The new commits contain a merge from "main" to "feature" as well as a merge from "feature" to "main".

I tried to reproduce it, so here's my process in doing that

The git log graph now looks like this (`git log --follow -m --graph --abbrev-commit --decorate --no-color --format=format:'%h - %aD (%ar)%d%n %s - %an' -- 'I🖤 Unicode.rst'`):
```
*   0a3b095 - Mon, 19 Jan 2026 18:24:34 +0100 (3 days ago) (HEAD, origin/merge-commits)
|\   Merge feature branch into main branch - Matthias Geier
| *   c919076 - Sat, 17 Jan 2026 17:20:51 +0100 (5 days ago)
| |\   Merge main branch into feature branch - Matthias Geier
| * | 5ce1357 - Thu, 15 Jan 2026 22:13:07 +0100 (7 days ago)
| | |  Add comment about "orphan" page - Matthias Geier
...
* 23d25d0 - Sun, 31 Jan 2021 21:52:36 +0100 (5 years ago) (origin/master, origin/HEAD, master)
|  Add source file with Unicode file name - Matthias Geier
```

So currently, it should trigger on  `0a3b095`  / `time6` (which is indeed what you have in your test https://github.com/mgeier/sphinx-last-updated-by-git/pull/107/changes#diff-14a4e8d7de7cf300aee26333cd759892856e6a5d5740a094da66a806854b6bddL25).

With the changes in this PR it should be triggered on `5ce1357` / `time4` without `--first-parent` (`git log --graph --abbrev-commit --decorate --no-color --format=format:'%h - %aD (%ar)%d%n %s - %an' -- 'I 🖤 Unicode.rst'`):
```
* 5ce1357 - Thu, 15 Jan 2026 22:13:07 +0100 (7 days ago)
|  Add comment about "orphan" page - Matthias Geier
* 23d25d0 - Sun, 31 Jan 2021 21:52:36 +0100 (5 years ago) (origin/master, origin/HEAD, master)
   Add source file with Unicode file name - Matthias Geier
```
And it should be triggered on `0a3b095` / `time6` with the `--first-parent` option (`git log --follow --first-parent --graph --abbrev-commit --decorate --no-color --format=format:'%h - %aD (%ar)%d%n %s - %an' -- 'I � Unicode.rst'`
```
* 0a3b095 - Mon, 19 Jan 2026 18:24:34 +0100 (3 days ago) (HEAD, origin/merge-commits)
|  Merge feature branch into main branch - Matthias Geier
...
* 23d25d0 - Sun, 31 Jan 2021 21:52:36 +0100 (5 years ago) (origin/master, origin/HEAD, master)
|  Add source file with Unicode file name - Matthias Geier
```

For the other files I agree with the new timestampe (time5) too.

So in this PR we need to adapt the tests:

- [ ] set expected result of unicode to time4 https://github.com/mgeier/sphinx-last-updated-by-git/pull/107/changes#diff-14a4e8d7de7cf300aee26333cd759892856e6a5d5740a094da66a806854b6bddL25
- [ ] Add tests for with --first-parent, unicode to test6. Others remain time5 because edits were all on main
- [ ] Remove commits from exclude tests: https://github.com/mgeier/sphinx-last-updated-by-git/pull/107/changes#diff-14a4e8d7de7cf300aee26333cd759892856e6a5d5740a094da66a806854b6bddR189 and https://github.com/mgeier/sphinx-last-updated-by-git/pull/107/changes#diff-14a4e8d7de7cf300aee26333cd759892856e6a5d5740a094da66a806854b6bddR198
